### PR TITLE
Add Icon diagram

### PR DESCRIPTION
**Pull request by brianegan**
Hey hey :) 

I'm helping to generate images for the Flutter Docs (both API docs and Website). I wanted to start small to make sure I'm getting this right! Therefore, I'm adding a Diagram for the Icon class that can be added to the Docs and Widget catalog.

Thanks for your help!

### CODE REVIEW

File: utils/diagram_generator/lib/main.dart [hunk 1]
```diff
@@ -62,6 +63,7 @@ Future<Null> main() async {
     CustomListItemDiagramStep(controller),
     FlowDiagramStep(controller),
     HeroesDiagramStep(controller),
+    IconDiagramStep(controller),
```
Comment (brianegan) [inline utils/diagram_generator/lib/main.dart]: Sorry about some of this formatting / import noise -- this is the only change I actually made. The rest is `dartfmt` doing its thing.

**Comment by brianegan** [comment]
Woohoo! Thanks for the review :)

**Comment by brianegan** [comment]
Oh, one more Q -- Is it Ok if I merge this myself, or would you prefer to merge these kind of changes in to keep track of everything?

**Comment by Piinks** [comment]
With everything green and an LGTM - you may do the honors and merge away! 🎉

**Comment by brianegan** [comment]
Woohoo -- thanks again for your help :)

**Pull request merged**